Search Filters, Card Pages, Sidebar

adrianAlday opened the pull request:
#11 
#12
#19
#20
#21
#22 
#23 
#24 

sidebar:

added env file to my files

added star icon when not signed in
star icon color on hover
animate star color

added loading text
loading text color on hover
animate text color

logged in circle color on hover
animate circle color

long first name wrapping

no photo letter color on hover
animate letter color

add arrow icon
animate arrow icon flip

add menu
rotated arrow icon for menu settings
menu item background and icon color on hover and active
animate menu
prevent menu animation onload
menuopen added to redux store

add settings link and page
add logout link

sidebar pages:

sidebar background color on hover
animate sidebar background color

icon default color  prop to inherit
sidebar icon color on active
sidebar icon color on hover
animate sidebar icon color

search:

animate only opacity on search form active
prevent submit and reload of search form

filter:

added filter icons
filter icons color on click
filter icons color on hover
initial state ui.selectedsuits
selector grabs ui.selectedsuits to filter array
created action toggle suit
dispatched action with component
reducer makes new state
filter map render looks at single component state instead of one for each suit

list:

change card list suit icon default size
suit icon fill on row hover
rows link to card pages

git:

merged pr with readme and reorganization
addressed conflicted changes
updated icon location references

query:

initial state ui.query
selector grabs ui.query to search array
created action change query
dispatched action with component
reducer makes new state
remove query from component state

card pages:

card pages added
spin icon added
routing for card pages
shallow routing for reversing card
prevented wrap of title
set vertical align of title
fixed animation classnames with hashes
hidden elements removed from dom
prevent animation onload

general:

pointer cursor change on clickable

fmapilis commented on redux/selectors/cards.js:
```diff
@@ -1,17 +1,20 @@
-    const majors = originalArray.filter( card => card.arcana === "major" );
-    const minors = originalArray.filter( card => card.arcana !== "major" );
+    const majors = filteredArray.filter( card => card.arcana === "major" );
+    const minors = filteredArray.filter( card => card.arcana !== "major" );
```
This is optional but a neat way to get this into one line would be with `lodash`'s `partition` function:

```
const [ majors, minors ] = _.partition( filteredArray, card => card.arcana === "major" );
```

adrianAlday replied:
lodash partition to sort major/minor arcana

fmapilis commented on redux/reducers/ui.js:
```diff
@@ -1,3 +1,38 @@
+
+    const suit = payload;
+    const index = state.selectedSuits.indexOf( suit );
+    const newSuits = state.selectedSuits.slice();
```
👍

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+  }
+
+  state = {
+    initialPosition: true,
```
This may need a better name. I don't immediately know what 'initial position" means if it's true or false

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+          }
+          .flip-up {
+            animation: flip-up 0.6s ease-in-out forwards;
+          }
```
we can probably replace a lot of this for `.card` with a css `transition` instead of handling multiple class names and animations

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+            transition: fill 0.3s ease-in;
+            animation-play-state: running;	
+          }
+          .spin-icon-orientation-hover {
```
What's the difference between this and `.spin-icon:hover`?

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+            transition: fill 0.3s ease-in-out;
+            animation-play-state: running;	
+          }
+          .card:hover ~ .spin-icon{ 
```
Since these are the same exact styles, we can group these with the `.spin-icon:hover` rule above

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+            cursor: alias;
+          }
+          .all-themes {
+          }
```
Let's get rid of any unused css rules

adrianAlday replied:
removed empty css classes

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,305 @@
+
+  render() {
+
+    let cardStatus = this.state.initialPosition ? ( this.state.reversed ? "start-down" : "start-up" ) : ( this.state.reversed ? "flip-down" : "flip-up" );
```
I generally try to avoid putting multiple nested ternary expressions together since it's kind of hard to read. The two below this share the same comparison logic, so we can probably group all three into something more readable.

adrianAlday replied:
nested ternarys removed

fmapilis commented on pages/card-finder.js:
```diff
@@ -0,0 +1,136 @@
+
+        <style jsx>{`
+            .search-bar {
+              width: calc(100% - 352px);
```
I'd prefer it if we used a flexbox since we have 5 elements of fixed widths (the filter icons) and 1 element to fill in the rest, which is a perfect use case for `flex-grow` and `flex-basis`

adrianAlday replied:
flexbox used for search bar width

fmapilis commented on pages/card-finder.js:
```diff
@@ -0,0 +1,136 @@
+  { icon: PentacleIcon, suit: "pentacles" },
+  { icon: SwordIcon, suit: "swords" },
+  { icon: WandIcon, suit: "wands" },
+];
```
This same array is in the card page. We should probably move this out somewhere into a shared file and import it in instead.

adrianAlday replied:
suitinfo added to common folder

fmapilis commented on components/user.js:
```diff
@@ -0,0 +1,292 @@
+
+    let menuStyle = this.state.menuOpen
+      ? { height: "144px" }
+      : { height: "0px" };
```
this should be handled by a class name + a css rule

adrianAlday replied:
menu open rules consolidated into css

fmapilis commented on components/layout.js:
```diff
@@ -1,116 +1,153 @@
+            <User activePage = { this.props.activePage } />
+            {pageInfo.map(( page, i ) =>
+              <Link prefetch href={ page.path } key={ i }>
+                <div className={ `side-bar-page ${page.title === this.props.activePage ? "side-bar-active-page" : ""}` }>
```
Let's try to keep redundancy to a minimum with class names – `side-bar-active-page` could just be `active`.

Then we can differentiate states between `.side-bar-page` and `.side-bar-page.active`

adrianAlday replied:
layout component css classnames streamlined

fmapilis commented on components/layout.js:
```diff
@@ -1,116 +1,153 @@
+            fill: #ffdb79 !important;
+            transition: all 0.1s ease-in-out;
+          }
+          .side-bar-active-page {
```
This has the same rules as above

adrianAlday replied:
page:hover and active css classes routed to same rules

adrianAlday commented:
summary of changes included in c689b2b:

sidebar:
menu open rules consolidated into css
layout component css classnames streamlined
page:hover and active css classes routed to same rules

search page:
suitinfo added to common folder
flexbox used for search bar width
lodash partition to sort major/minor arcana
new card database updated in from master

card page:
connections updated for new card database schema
removed empty css classes
url query changed to "reversed"
string to boolean logic placed in get initial props
spin icon cursor css fixed
spin-icon:hover and card:hover ~ .spin-icon routed to same rules
removed clicking orientation word in title to reverse card
added card description and crystal ball divider
switch from css animations to transitions 
divs remain, only content and classes change based on state
nested ternarys removed
"neverclicked" state remains to prevent transition on load
card page split into left and right column divs
next and previous card links added

fmapilis commented on pages/card.js:
```diff
@@ -0,0 +1,252 @@
+              <BallIcon />
+            </div>
+            <div className={ `description fade ${ transitionClass }` }>
+              { this.props.pageCard.meaning[ this.state.reversedOnTransitionEnd ? "reversed" : "upright" ] }
```
The meanings are strings that have new line separators `\n` (more specifically, paragraphs are separated by two of them, so `\n\n`) that aren't showing as linebreaks on the page.

We'll want to split these into their own paragraphs. You could do something like:
```javascript
meaning.split("\n\n").map(( paragraph, i ) => <p key={ i }>{paragraph}</p> )
```

fmapilis replied:
Oh, and this goes for descriptions too. I think most descriptions are just one paragraph, but there are some like The Magician that have multiple.

adrianAlday replied:
description and meanings split into p elements at \n\ns using split and map. thats really neat!

fmapilis commented on components/layout.js:
```diff
@@ -1,116 +1,143 @@
+            .corner:hover .corner-title {
+              margin: 30px 0px 30px 14px;
+              letter-spacing: 2.5px;
+              transition: margin 0.3s ease-in-out, letter-spacing 0.3s ease-in-out,;
```
You only need the transition on `.corner-title` since `.corner:hover .corner-title` will inherit anything that `.corner-title` has (and then overwrite anything, like `margin` and `letter-spacing` in this case.

adrianAlday replied:
redundant css line removed

fmapilis commented on components/layout.js:
```diff
@@ -1,116 +1,143 @@
+              font-size: 20px;
+              letter-spacing: 1.5px;
+              color: #ffdb79;
+              transition: margin 0.3s ease-in-out, letter-spacing 0.3s ease-in-out,;
```
This animation isn't working because of the stray `,` at the end of the rule.

adrianAlday replied:
css typo fixed

fmapilis commented on redux/store.js:
```diff
@@ -9,14 +9,18 @@ const initialState = {
   },
-  user: {},
+  ui: {
+    menuOpen: false,
```
We should use a better name for this - `menuOpen` is pretty vague, but really it just refers to the sidebar user menu. There are probably going to be a lot of other menus down the line.

adrianAlday replied:
menuOpen renamed to userAccountMenuOpen

adrianAlday commented:
layout:
redundant css line removed
css typo fixed

various:
menuOpen renamed to userAccountMenuOpen

card page:
description and meanings split into p elements at \n\ns using split and map. thats really neat!
